src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,149 @@
+                ? 0
+                : Math.min(numExpectedComments - MAX_COMMENTS_PER_TOKEN, MAX_COMMENTS_PER_TOKEN),
+            nextPageToken,
+            SECOND_TOKEN_COMMENT);
+    when(mockedCommentThreadList.execute()).thenReturn(firstResponse, secondResponse);
+    when(mockedYoutube.commentThreads().list(anyString())).thenReturn(mockedCommentThreadList);
+    commentRetriever = new YouTubeCommentRetriever(mockedYoutube);
+  }
+
+  // Creates list of comments of desired length.
+  // Page token is needed for iteration but its contents are irrelevant.
+  private CommentThreadListResponse mockThreadListResponse(
+      int desiredComments, String nextPageToken, String commentContent) {
+    CommentThread mockThread = dummyCommentThread(commentContent);
+    return new CommentThreadListResponse()
+        .setItems(Collections.nCopies(desiredComments, mockThread))
+        .setNextPageToken(nextPageToken);
+  }
+
+  private CommentThread dummyCommentThread(String comment) {
+    // This has to be done because for some reason it's impossible to directly change a comment's
+    // content
+    CommentThread mockedCommentThread = mock(CommentThread.class, RETURNS_DEEP_STUBS);
+    when(mockedCommentThread.getSnippet().getTopLevelComment().getSnippet().getTextDisplay())
+        .thenReturn(comment);
+    return mockedCommentThread;
+  }
+
+  // The simplest case: extract 100 comments from a video with more than 100 comments
+  @Test
+  public void testDefaultBehaviour() throws Exception {
+    setUp(HUNDRED);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, HUNDRED);
+    Assert.assertEquals(comments.size(), HUNDRED);
+  }
+
+  // Extract more than 100 comments from a video
+  @Test
+  public void testExcessHundredComments() throws Exception {
+    setUp(EXCESS_HUNDRED, NEXT_PAGE_TOKEN);
+    List<CommentThread> comments =
+        commentRetriever.retrieveComments(POPULAR_VIDEO_URL, EXCESS_HUNDRED);
+    Assert.assertEquals(comments.size(), EXCESS_HUNDRED);
+  }
+
+  // Only load the comments that are in a video, without crashing
+  @Test
+  public void doesNotAttemptRetrieveExcess() throws Exception {
+    setUp(1);
+    List<CommentThread> comments =
+        commentRetriever.retrieveComments(UNPOPULAR_VIDEO_URL, EXCESS_HUNDRED);
+    Assert.assertEquals(comments.size(), 1);
+  }
+
+  // Retrieve a specific amount of comments less than 100
+  @Test
+  public void retrievesSpecificNumComments() throws Exception {
+    setUp(12);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, 12);
+    Assert.assertEquals(comments.size(), 12);
+  }
+
+  // Retrieve a specific amount of comments more than 100
+  @Test
+  public void retrievesSpecificNumCommentsExcessHundred() throws Exception {
+    setUp(120, NEXT_PAGE_TOKEN);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, 120);
+    Assert.assertEquals(comments.size(), 120);
+  }
```

Comment:
This test seems to be testing the same thing as testExcessHundredComments()? Remove one please.